src/Sanitizers.cmake

```diff
@@ -86,8 +86,17 @@ function(
               "Using MSVC sanitizers requires setting the MSVC environment before building the project. Please manually open the MSVC command prompt and rebuild the project."
           )
         endif()
+        if(POLICY CMP0141)
+          if(CMAKE_MSVC_DEBUG_INFORMATION_FORMAT STREQUAL "EditAndContinue")
+            set_target_properties(
+              ${_project_name} PROPERTIES MSVC_DEBUG_INFORMATION_FORMAT ProgramDatabase
+            )
+          endif()
+        else()
+          target_compile_options(${_project_name} INTERFACE /Zi)
+        endif()
         target_compile_options(
-          ${_project_name} INTERFACE /fsanitize=${LIST_OF_SANITIZERS} /Zi /INCREMENTAL:NO
+          ${_project_name} INTERFACE /fsanitize=${LIST_OF_SANITIZERS} /INCREMENTAL:NO
```

Comment:
Would this break the default sanitizer experience? We want the sanitizer to work by default in MSVC.

Reply:
I don't think so, from version 3.25 CMake sets `MSVC_DEBUG_INFORMATION_FORMAT` to `$<$<CONFIG:Debug,RelWithDebInfo>:ProgramDatabase>`, (see: https://cmake.org/cmake/help/latest/variable/CMAKE_MSVC_DEBUG_INFORMATION_FORMAT.html) it might break for `Release/MinSizeRel` builds as no debug info is generated by default.
We could set `MSVC_DEBUG_INFORMATION_FORMAT` to `ProgramDatabase` unconditionally, users can still override it in their own targets, but if they have explicitly set `CMAKE_MSVC_DEBUG_INFORMATION_FORMAT` to a compatible value we should not override it.

Reply:
I just want to make sure we don't have a regression in the default experience. So as long as that's the case, this change should be OK.